internal/controller/volumereplicationgroup_controller.go

```diff
@@ -1236,9 +1236,10 @@ func (v *VRGInstance) updateStatusState() {
 	dataReadyCondition := findCondition(v.instance.Status.Conditions, VRGConditionTypeDataReady)
 	if dataReadyCondition == nil {
 		// VRG is exclusively using volsync
-		if v.instance.Spec.ReplicationState == ramendrv1alpha1.Secondary &&
-			len(v.instance.Spec.VolSync.RDSpec) > 0 {
-			v.instance.Status.State = ramendrv1alpha1.SecondaryState
+		if v.instance.Spec.ReplicationState == ramendrv1alpha1.Secondary {
```

Comment:
I am mostly certain that the code does not enter this if loop. This is a remnant of older code pre-aggregation changes.